src/layout/layout.tsx

```diff
@@ -1,49 +1,64 @@
-import { Drawer, List, ListItem, ListItemButton, ListItemIcon, Toolbar, ListItemText, Box } from '@mui/material';
+import React from 'react';
+import { Toolbar, List, ListItem, ListItemText, CssBaseline, Box, ListItemButton, ListItemIcon } from '@mui/material';
 import { Settings as ConfigIcon, Description as SchemaIcon } from '@mui/icons-material';
-import { useNavigate } from 'react-router-dom';
 import { routes } from '../routing/routes';
+import { useNavigate } from 'react-router-dom';
+import Styles from './layout.module.scss';
 
 type LayoutProps = {
   children: React.ReactNode;
 };
 
 type DrawerItem = {
+  id: number;
   text: string;
   icon: React.ReactNode;
   onClick: () => void;
 };
-
 export const Layout: React.FC<LayoutProps> = ({ children }) => {
   const navigate = useNavigate();
+  const [selectedButtonIndex, setSelectedButtonIndex] = React.useState<number>(0);
```

Comment:
maybe you can use something like this instead of selecting manually? https://mui.com/material-ui/integrations/routing/#list we can check this together

Reply:
thanks